### PR TITLE
Reset position of vsc with hotkey

### PR DESCRIPTION
**Pull request by jaywoozy**
Added a hot key ("H" for "home") to move the video speed controller back to the upper left corner to avoid losing the controller due to varying video resizings, or for when it gets stuck

### CODE REVIEW

File: inject.js [hunk 1]
```diff
@@ -613,6 +623,21 @@
     parentElement.addEventListener('mousemove', startDragging);
   }
 
+  function handleReposition(video, controller) { // 481_NEW: handling reposition
+    const shadowController = controller.shadowRoot.querySelector('#controller');
+    // Find nearest parent of same size as video parent.
+    var parentElement = controller.parentElement;
```
Comment (igrigorik) [inline inject.js]: Silly question, why is this block necessary? We don't use parentelElement below?

File: options.html [hunk 2]
```diff
@@ -82,6 +82,16 @@ <h3>Shortcuts</h3>
                 <option value="false">Do not disable website key bindings</option>
                 <option value="true">Disable websites key bindings</option>
             </select></div>
+        <div class="row customs" id="reposition">
+            <select class="customDo">
+                <option value="reposition">Reset controller position</option>
+            </select>
+            <input class="customKey" type="text" value="" placeholder="press a key">
+            <input class="customValue" type="text" placeholder="value (0.10)">
```
Comment (igrigorik) [inline options.html]: Is that the right placeholder value?

**Comment by igrigorik** [comment]
WDYT of extending this to enable https://github.com/igrigorik/videospeed/issues/423? We could provide top left, bottom left as options, maybe?

**Comment by ChadBailey** [comment]
@jaywoozy Could I have you take a look at this one see if you can respond to some of @igrigorik 's feedback?

File: inject.js (hunk 1 above)
Comment (jaywoozy) [reply inject.js]: Sorry the the super late response, I forgot about this contribution. This does seem redundant, we could probably just use the parentElement below as you've said

File: options.html (hunk 2 above)
Comment (jaywoozy) [reply options.html]: I was unsure about what to put down as the placeholder value for this, so I just put in anything for the time being. Not sure if this needs any placeholder at all, though

**Comment by jaywoozy** [comment]
@ChadBailey @igrigorik So sorry for the super late responses! I almost forgot I made these contributions. I've responded to some of them now, let me know if there's anything else!

**Comment by ChadBailey** [comment]
No problem! I'm glad you responded, and thanks for your contribution!

**Comment by jaywoozy** [comment]
No problem! Thank you guys for making this, it's my absolute favorite chrome extension of all time! Keep up the great work

**Comment by ChadBailey** [comment]
Alright, once we get the current set of bugs fixed up (not many, expected by this weekend) we should be able to get this merged in. I've slated this for 0.7 release

**Comment by igrigorik** [comment]
@jaywoozy @ChadBailey either of you willing to resolve the merge conflicts to help land this? :)

**Comment by jaywoozy** [comment]
Yes I’m busy now but I can take a look later tonight or tomorrow!

**Comment by igrigorik** [comment]
Ty!

File: inject.js (hunk 1 above)
Comment (jaywoozy) [reply inject.js:882]: Is there a particular way that you'd like me to refactor this code?

File: options.html (hunk 2 above)
Comment (jaywoozy) [reply options.html]: @igrigorik how do you feel about keeping that "dummy" value there? Or should I change it somehow

File: options.js [hunk 3]
```diff
@@ -15,7 +15,8 @@ var tcDefaults = {
     { action: "rewind", key: 90, value: 10, force: false, predefined: true }, // Z
     { action: "advance", key: 88, value: 10, force: false, predefined: true }, // X
     { action: "reset", key: 82, value: 1, force: false, predefined: true }, // R
-    { action: "fast", key: 71, value: 1.8, force: false, predefined: true } // G
+    { action: "fast", key: 71, value: 1.8, force: false, predefined: true }, // G
+    {action: "reposition", key: 72, value: 0, force: false, predefined: true} // H
```
Comment (igrigorik) [inline options.js:19]: knit, missing space

File: options.js [hunk 4]
```diff
@@ -126,7 +127,7 @@ function updateCustomShortcutInputText(inputItem, keyCode) {
 }
 
 // List of custom actions for which customValue should be disabled
-var customActionsNoValues = ["pause", "muted", "mark", "jump", "display"];
+var customActionsNoValues = ["pause", "muted", "mark", "jump", "display","reposition"];
```
Comment (igrigorik) [inline options.js:130]: space between values.

(these should be caught by your formatter / linter :))

File: inject.js [hunk 5]
```diff
@@ -868,11 +876,25 @@ function handleDrag(video, controller, e) {
...
+    while (parentElement.parentNode &&
+      parentElement.parentNode.offsetHeight === parentElement.offsetHeight &&
+      parentElement.parentNode.offsetWidth === parentElement.offsetWidth) {
+      parentElement = parentElement.parentNode;
+    }
+
+    let style = shadowController.style;
```
Comment (igrigorik) [inline inject.js:889]: Wait, now that I'm looking at this.. it's tricky, I don't think 0 is the right value and, in fact, we have custom positions on various sites: https://github.com/igrigorik/videospeed/blob/master/inject.css

Hmm?

**Comment by ankitsingh101** [comment]
Is it functioning properly i have tired the shortcut H but not working

**Comment by igrigorik** [comment]
@jaywoozy @ChadBailey what's the status here, do we still intend / want to land this? If so, needs a rebase and a few cleanup updates based on last review.

**Comment by ChadBailey** [comment]
I'm not opposed to merging, but haven't had the chance to personally test

**Comment by igrigorik** [comment]
Fair point. I'll close this, @ankitsingh101 thanks for your work here and please feel free to reopen if you feel strongly otherwise / have a compelling counter. :)

**Pull request closed without merging**